File: app/src/main/java/cz/marvincz/rssnotifier/model/ChannelWithItems.java
package cz.marvincz.rssnotifier.model;

import android.net.Uri;

import java.util.List;
import java.util.Objects;
import java.util.Optional;
import java.util.Set;

import androidx.room.Relation;
import cz.marvincz.rssnotifier.model.RssChannel;
import cz.marvincz.rssnotifier.model.RssItem;
import cz.marvincz.xmlpullparserconverter.annotation.XmlElement;
import cz.marvincz.xmlpullparserconverter.annotation.XmlRootElement;
import paperparcel.PaperParcel;

@XmlRootElement(name = "rss/channel")
public class ChannelWithItems extends RssChannel {
    @Relation(parentColumn = "accessUrl", entityColumn = "channelUrl")
    @XmlElement(name = "item")
    public List<RssItem> items;

    public ChannelWithItems fixUrl(Uri url) {
        accessUrl = Objects.requireNonNull(url);
        Optional.ofNullable(items)
                .ifPresent(list -> list.forEach(i -> i.channelUrl = url));
        return this;
    }
}

File: app/src/main/java/cz/marvincz/rssnotifier/model/ChannelWithItems.kt
package cz.marvincz.rssnotifier.model

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml
import com.tickaroo.tikxml.converter.htmlescape.HtmlEscapeStringConverter

@Xml(name = "channel")
data class ChannelWithItems(
        @Element val items: List<RssItem> = emptyList(),
        @PropertyElement val link: String? = null,
        @PropertyElement(converter = HtmlEscapeStringConverter::class) val title: String? = null,
        @PropertyElement(converter = HtmlEscapeStringConverter::class) val description: String? = null
)